src/gui/entry/EntryView.cpp

```diff
@@ -269,159 +203,116 @@ Entry* EntryView::entryFromIndex(const QModelIndex& index)
 
-    /*
-     * Always set sorting to column 'Group', as it does not feel right to have
-     * the last known sort configuration of search view restored by 'Database
-     * WidgetStateSync', which is what happens without this
-     */
+    // Always set sorting to column 'Group', as it does not feel right to
+    // have the last known sort configuration of search view restored by
+    // 'DatabaseWidgetStateSync', which is what happens without this
     m_sortModel->sort(EntryModel::ParentGroup, Qt::AscendingOrder);
     sortByColumn(EntryModel::ParentGroup, Qt::AscendingOrder);
 
     m_inSearchMode = true;
 }
 
 /**
- * @author Fonic <https://github.com/fonic>
  * Get current state of 'Hide Usernames' setting (NOTE: just pass-through for
  * m_model)
  */
-bool EntryView::hideUsernames() const
+bool EntryView::isUsernamesHidden() const
 {
-    return m_model->hideUsernames();
+    return m_model->isUsernamesHidden();
 }
 
 /**
- * @author Fonic <https://github.com/fonic>
  * Set state of 'Hide Usernames' setting (NOTE: just pass-through for m_model)
  */
-void EntryView::setHideUsernames(const bool hide)
+void EntryView::setUsernamesHidden(const bool hide)
 {
-    m_model->setHideUsernames(hide);
+    m_model->setUsernamesHidden(hide);
 }
 
 /**
- * @author Fonic <https://github.com/fonic>
  * Get current state of 'Hide Passwords' setting (NOTE: just pass-through for
  * m_model)
  */
-bool EntryView::hidePasswords() const
+bool EntryView::isPasswordsHidden() const
 {
-    return m_model->hidePasswords();
+    return m_model->isPasswordsHidden();
 }
 
 /**
- * @author Fonic <https://github.com/fonic>
  * Set state of 'Hide Passwords' setting (NOTE: just pass-through for m_model)
  */
-void EntryView::setHidePasswords(const bool hide)
+void EntryView::setPasswordsHidden(const bool hide)
 {
-    m_model->setHidePasswords(hide);
+    m_model->setPasswordsHidden(hide);
 }
 
 /**
- * @author Fonic <https://github.com/fonic>
- * Get current state of view
+ * Get current view state
  */
 QByteArray EntryView::viewState() const
 {
     return header()->saveState();
 }
 
 /**
- * @author Fonic <https://github.com/fonic>
- * Set state of entry view view
+ * Set view state
  */
 bool EntryView::setViewState(const QByteArray& state) const
 {
     return header()->restoreState(state);
 }
 
 /**
- * @author Fonic <https://github.com/fonic>
  * Sync checkable menu actions to current state and display header context
  * menu at specified position
  */
 void EntryView::showHeaderMenu(const QPoint& position)
 {
-    /* Sync checked state of menu actions to current state of view */
-    m_hideUsernamesAction->setChecked(m_model->hideUsernames());
-    m_hidePasswordsAction->setChecked(m_model->hidePasswords());
+    m_hideUsernamesAction->setChecked(m_model->isUsernamesHidden());
+    m_hidePasswordsAction->setChecked(m_model->isPasswordsHidden());
     foreach (QAction *action, m_columnActions->actions()) {
```

Comment:
You still need to replace this with a range-base for loop.

Reply:
Why? foreach is recommended for Qt containers (in this case: QList< QAction* >).

Reply:
No, it's not. foreach has been deprecated for a while.

Reply:
As I recall that was planned, but never done. As of Qt 5.10, it is still recommended, see documentation.

Reply:
Nope. The prose documentation about containers still mentions it, but the QtGlobal reference lists it as deprecated since 5.7: https://doc-snapshots.qt.io/qt5-dev/qtglobal.html#Q_FOREACH

Reply:
Interesting, that's new to me. Can't get the range-based loop to work with asConst, though.

Reply:
```
for (auto& action: asConst(m_columnActions->actions()) {

}
```
Should work. Use the asConst() implementation from core/global.h instead of the Qt-provided qAsConst() one, because that one's not available in our targeted minimum version.

Reply:
Already tried that and all other variants I could think of. What you suggested gives:
src/gui/entry/EntryView.cpp:294:58: error: use of deleted function ‘void asConst(const T&&) [with T = QList<QAction*>]’

Reply:
Hav you added `#include "core/Global.h` ?

Reply:
yes

Reply:
I see, the move signature is disabled, so it's not accepting an rvalue. In that case, just save the result of m_columnActions->actions() in a const variable and iterate over that. Then you also don't need asConst() at all.

Reply:
Done